pkg/auth/handler/webapp/renderer.go

```diff
@@ -51,3 +54,29 @@ func (r *HTMLRenderer) Render(w http.ResponseWriter, req *http.Request, template
 	w.Header().Set("Content-Length", strconv.Itoa(len(body)))
 	w.Write(body)
 }
+
+func (r *HTMLRenderer) RenderAttachment(w http.ResponseWriter, req *http.Request, templateType config.TemplateItemType, data interface{}, filename string) {
+	r.Logger.WithFields(map[string]interface{}{
+		"data": data,
+	}).Debug("render with data")
+
+	preferredLanguageTags := intl.GetPreferredLanguageTags(req.Context())
+	out, err := r.TemplateEngine.WithValidatorOptions(
+		template.AllowRangeNode(true),
+		template.AllowTemplateNode(true),
+		template.AllowDeclaration(true),
+		template.MaxDepth(15),
+	).WithPreferredLanguageTags(preferredLanguageTags).RenderTemplate(
+		templateType,
+		data,
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	body := []byte(out)
+	w.Header().Set("Content-Type", "text/plain; charset=utf-8")
+	w.Header().Set("Content-Length", strconv.Itoa(len(body)))
+	w.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=\"%s\"", filename))
```

Comment:
Let's assume `filename` does not need escaping, since we're not passing dynamic data in filename?